.travis.yml

```diff
@@ -3,77 +3,147 @@ language: scala
 before_install:
   # See https://github.com/travis-ci/travis-ci/issues/4629#issuecomment-239493916
   - rm ~/.m2/settings.xml
-  - curl -Ls https://git.io/jabba | bash && . ~/.jabba/jabba.sh
-
-install: jabba install adopt@1.8.192-12 && jabba use $_ && java -version
```

Comment:
Keep this, but make it

```yaml
install: jabba install "$TRAVIS_JDK" && jabba use $_ && java -Xmx32m -version
```

with:

* TRAVIS_JDK defaulted as `TRAVIS_JDK=adopt@1.8.202-08` in `env.matrix`, and
* `env: TRAVIS_JDK=adopt@1.11.0-2` in the `test-java11` jobs.

So like Play's setup: https://github.com/playframework/playframework/blob/master/.travis.yml

At this point in time I'm not a fan of the creative indirection of `bin/jabbaLib` functions... but that's just my gut reaction.  Perhaps I could come around (and maybe even promote it) 😄

Reply:
I tried to bring this `.travis.yml` close to the one in playframework/play-samples but this uses `stages` for deployment jobs and my head exploded a bit.